flocker/control/test/test_schemas.py

```diff
@@ -98,6 +98,23 @@
                 {'internal': 80, 'external': 8080},
             ]
         },
+        # Environment given but not a dict
+        {
+            'host': '192.168.0.3',
+            'image': 'postgres',
+            'name': 'postgres',
+            'environment': 'x=y'
+        },
+        # Environment given but at least one entry is not a string
+        {
+            'host': '192.168.0.3',
+            'image': 'postgres',
+            'name': 'postgres',
+            'environment': {
+                'POSTGRES_USER': 'admin',
+                'POSTGRES_VERSION': 9.4
+            }
+        },
```

Comment:
I think the schema limits the key / value to 256 characters. Add a test for that.